### DISCUSSION TITLE
[Detector Support]: Halio8 nmo longer finds things like robot_lawnmower and waste_bin

### DISCUSSION THREAD

**Discussion by ljmerza** [blakeblackshear/frigate#20618] [Detector Support]
### Describe the problem you are having

I switch from coral to hailo8 and latest plus models but now im only detecting things like person, cat, dog. It doesn't detect waste_bin and robot_lawnmower at all like the coral did

### Version

0.16.2-tensorrt

### Frigate config file

```yaml
back_yard_side_camera:
    ffmpeg:
      inputs:
        - path: 
            rtsp://admin:{FRIGATE_AMCREST_PASSWORD}@192.168.1.140:554/cam/realmonitor?channel=1&subtype=0
          roles:
            - record
        - path: 
            rtsp://admin:{FRIGATE_AMCREST_PASSWORD}@192.168.1.140:554/cam/realmonitor?channel=1&subtype=1
          roles:
            - detect
    detect:
      width: 704
      height: 480
      fps: 5
    objects:
      track:
        - person
        - dog
        - cat
        - bird
        - deer
        - rabbit
        - squirrel
        - waste_bin
        - face
        - fox
        - raccoon
        - robot_lawnmower
        - possum

detectors:
  hailo:
    type: hailo8l
    device: PCIe
    model:
      # yolov9s 640x640 hailo8
      path: plus://8a66079f4db4577a78d2f12c9cfc2814
```

### docker-compose file or Docker CLI command

```yaml
frigate:
    image: ghcr.io/blakeblackshear/frigate:0.16.2-tensorrt
    container_name: frigate
    privileged: true
    restart: unless-stopped
    shm_size: 1024mb
    env_file: ./env/frigate.env
    cap_add:
      - CAP_PERFMON
    devices:
      - /dev/dri:/dev/dri # For intel hwaccel, needs to be updated for your hardware
      - /dev/hailo0:/dev/hailo0 # Passes a USB Hailo, follow driver instructions here https://docs.hailo.ai/hailo-8-linux-getting-started-guide/getting-started.html#2-install-the-hailo-linux-drivers
      - /dev/dri/renderD128:/dev/dri/renderD128 # For intel hwaccel, needs to be updated for your hardware
    # - /dev/bus/usb:/dev/bus/usb # Passes the USB Coral, needs to be modified for other versions
    # - /dev/apex_0:/dev/apex_0 # Passes a PCIe Coral, follow driver instructions here https://coral.ai/docs/m2/get-started/#2a-on-linux
    volumes:
      - ./volumes/frigate/config:/config
      - /media/cubxi/nvr/frigate:/media/frigate
      - /etc/localtime:/etc/localtime:ro
      - type: tmpfs
        target: /tmp/cache
        tmpfs:
          size: 4000000000
    ports:
      - 5001:5000
    networks:
      camera_network:
    deploy:
      resources:
        reservations:
          devices:
            - driver: nvidia
              count: all
              capabilities: [gpu]
```

### Relevant Frigate log output

```shell
no logs it doesnt find anything
```

### Install method

Home Assistant Add-on

### Object Detector

Other

### Screenshots of the Frigate UI's System metrics pages

<img width="1880" height="874" alt="image" src="https://github.com/user-attachments/assets/400d8c31-5022-40ee-b827-f0a2c91b8b4e" />

**Comment by inkeep-ai-assistant** [comment]
The issue you're experiencing is related to the difference between the default Frigate models and Frigate+ models. When you switched from Coral to Hailo8 with Frigate+ models, the available object types changed.

Frigate+ models support a specific set of objects that differs from the default models [(1)](https://docs.frigate.video/plus). According to the documentation, the available label types in Frigate+ models include `waste_bin` and `robot_lawnmower` under the "Other" category [(1)](https://docs.frigate.video/plus):

- **Other**: `package`, `waste_bin`, `bbq_grill`, `robot_lawnmower`, `umbrella` [(1)](https://docs.frigate.video/plus)

Since you're using a Frigate+ model (indicated by your `path: plus://8a66079f4db4577a78d2f12c9cfc2814` configuration), these objects should be supported. However, there are a few things to check:

1. **Verify your model configuration**: You're using a YOLOv9 model which is supported on Hailo devices [(1)](https://docs.frigate.video/plus). For Hailo8/Hailo8L hardware, `yolov9` is the recommended model type [(1)](https://docs.frigate.video/plus).

2. **Check your object tracking configuration**: Make sure you have the objects properly configured in your `objects.track` list, which you do have correctly set up.

3. **Consider threshold adjustments**: The documentation notes that when switching to YOLOv9, you may need to adjust your thresholds for some objects [(1)](https://docs.frigate.video/plus).

4. **Model training**: Frigate+ models improve through fine-tuning with your specific images. The documentation suggests submitting diverse images and both true and false positives to improve model accuracy [(1)](https://docs.frigate.video/plus). You should provide at least 100 verified images per camera with varying conditions [(1)](https://docs.frigate.video/plus).

Since you were detecting these objects with Coral but not with Hailo8, this suggests the Frigate+ model may need more training data for your specific environment to properly detect `waste_bin` and `robot_lawnmower` objects.

📚 **Sources:**
1. [Models](https://docs.frigate.video/plus)

---

Have another question? Just tag [@inkeep](https://inkeep.com).

**Reply by ljmerza** [reply]
I used frigate plus models for coral and it worked fine

**Comment by NickM-27** [comment]
Your config is invalid

```
detectors:
  hailo:
    type: hailo8l
    device: PCIe
    model:
      # yolov9s 640x640 hailo8
      path: plus://8a66079f4db4577a78d2f12c9cfc2814

```

needs to be

```
detectors:
  hailo:
    type: hailo8l
    device: PCIe

model:
  # yolov9s 640x640 hailo8
  path: plus://8a66079f4db4577a78d2f12c9cfc2814
```
